ocs_samples/library_samples/Java/ocs_sample_library_preview/src/main/java/com/github/osisoft/ocs_sample_library_preview/sds/StreamsClient.java

```diff
@@ -927,6 +928,56 @@ public String getWindowValues(String tenantId, String namespaceId, String stream
         return jsonResults.toString();
     }
 
+    /**
+     * gets sampled values from the stream
+     * @param tenantId tenant to work under
+     * @param namespaceId namespace within tenant
+     * @param streamId name of stream to get data from
+     * @param startIndex starting index
+     * @param endIndex ending index
+     * @param intervals number of intervals to run sample
+     * @param sampleBy property to sample by
+     * @return
+     * @throws SdsError errors that may occur
+     */
+    public String getSampledValues(String tenantId, String namespaceId, String streamId, String startIndex, String endIndex, int intervals, String sampleBy){
+        URL url = null;
+        HttpURLConnection urlConnection = null;
+        String inputLine;
+        StringBuffer response = new StringBuffer();
+        try{
+            url = new URL(baseUrl + getSampledValuesQuery.replace("{apiVersion}", apiVersion).replace("{tenantId}", tenantId).replace("{namespaceId}", namespaceId)
+            .replace("{streamId}", streamId).replace("{startIndex}", startIndex).replace("{endIndex}", endIndex)
+            .replace("{intervals}", "" + intervals).replace("{sampleBy}", sampleBy));
+            urlConnection = baseClient.getConnection(url, "GET");
+        }
+         catch (MalformedURLException mal) {
+            System.out.println("MalformedURLException");
+        } catch (IllegalStateException e) {
+            e.getMessage();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        
+        try{
+            int httpResult = urlConnection.getResponseCode();
+            if(httpResult != HttpURLConnection.HTTP_OK){
+                throw new SdsError(urlConnection, "get sampled values request failed");
+            }
+            BufferedReader in = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
+
+            while((inputLine = in.readLine()) != null){
+                response.append(inputLine);
+            }
+            in.close();
+        } catch (SdsError sdsError) {
+            sdsError.print();
+            //throw sdsError;
```

Comment:
should this be here?

Reply:
I believe that should be there, just not commented out. It was exhibiting some inexplicable behavior for that one instance of error while I was working on it even though the same exact call appears in other places without issue.

Reply:
So i figured why it's failing here you need to add  'throws SdsError' after method declaration and before method body. Try it and let me know.